crates/parcel_monitoring/src/sentry_integration.rs

```diff
@@ -0,0 +1,173 @@
+//! This module configures `sentry` to report panics.
+//!
+//! Sentry is disabled by default.
+use std::collections::HashMap;
+
+use anyhow::anyhow;
+use sentry::configure_scope;
+use sentry::ClientInitGuard;
+use serde_json::Value;
+use whoami::username;
+
+use crate::from_env::{optional_var, required_var, FromEnvError};
+
+fn value_to_string(value: &Value) -> String {
+  match value {
+    Value::String(inner) => inner.clone(),
+    other => other.to_string(),
+  }
+}
+
+#[derive(Debug)]
+pub struct SentryOptions {
+  pub sentry_dsn: String,
+  pub sentry_tags: HashMap<String, String>,
+}
+
+impl SentryOptions {
+  pub fn from_env() -> Result<Option<SentryOptions>, FromEnvError> {
+    if optional_var("PARCEL_ENABLE_SENTRY").is_none() {
+      return Ok(None);
+    }
+
+    let sentry_tags = match optional_var("PARCEL_SENTRY_TAGS") {
+      Some(tags_string) => {
+        let sentry_tags =
+          serde_json::from_str::<HashMap<String, Value>>(&tags_string).map_err(|err| {
+            FromEnvError::InvalidKey(
+              String::from("PARCEL_SENTRY_TAGS"),
+              anyhow!("Invalid JSON on tags: {:#?}", err),
+            )
+          })?;
+        let sentry_tags: HashMap<String, String> = sentry_tags
+          .iter()
+          .map(|(k, v)| (k.clone(), value_to_string(v)))
+          .collect::<HashMap<String, String>>();
+
+        sentry_tags
+      }
+      None => HashMap::new(),
+    };
+
+    Ok(Some(SentryOptions {
+      sentry_dsn: required_var("PARCEL_SENTRY_DSN")?,
+      sentry_tags,
+    }))
+  }
+}
+
+/// Initializes sentry for error monitoring. This will report panics.
+pub fn init_sentry(options: SentryOptions) -> anyhow::Result<ClientInitGuard> {
+  tracing::info!("Initialising Sentry in rust...");
+
+  let mut sentry_client_options = sentry::ClientOptions {
+    ..Default::default()
+  };
+
+  if let Some(release) = options.sentry_tags.get("release") {
+    sentry_client_options.release = Some(release.to_string().into());
+  }
+  if let Some(environment) = options.sentry_tags.get("environment") {
+    sentry_client_options.environment = Some(environment.to_string().into());
+  }
+  if let Some(debug) = options.sentry_tags.get("debug") {
+    sentry_client_options.debug = debug.to_string() == "true";
+  }
+
+  let guard = sentry::init((options.sentry_dsn, sentry_client_options));
+
+  configure_scope(|scope| {
+    scope.set_user(Some(sentry::User {
+      username: Some(username()),
+      ..Default::default()
+    }));
+  });
+
+  configure_scope(|scope| {
```

Comment:
Can't you collapse this in the previous `configure_scope`?

Reply:
Done ;